src/ncp/ncp_base.cpp

```diff
@@ -868,9 +915,15 @@ void NcpBase::CommandHandler_NOOP(uint8_t header, unsigned int command, const ui
 
 void NcpBase::CommandHandler_RESET(uint8_t header, unsigned int command, const uint8_t *arg_ptr, uint16_t arg_len)
 {
-    // TODO: Figure out how to actually perform a reset.
-    otInit();
-    SendLastStatus(SPINEL_HEADER_FLAG | SPINEL_HEADER_IID_0, SPINEL_STATUS_RESET_SOFTWARE);
+    otPlatReset();
+
+    // We only get to this point if the
```

Comment:
Here we set all bits of `mChangedFlags`? For example, we are setting both `OT_IP6_ADDRESS_ADDED` and `OT_IP6_ADDRESS_REMOVED` flags. This works ok now (we clear them all together as we send stuff out).  I feel we may need to be careful about the value we set here when we add new flags (e.g., if we handle INSERTED case).